File: src/main/java/com/dorpine/model/Track.java
package com.dorpine.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;

@JsonIgnoreProperties(ignoreUnknown = true)
public class Track {
    private String id;
    private String title;
    private String artist;
    private String genre;
    @JsonProperty("durationSeconds")
    private int durationSeconds;
    @JsonProperty("coverUrl")
    private String coverUrl;
    @JsonProperty("spotifyId")
    private String spotifyId;
    @JsonProperty("audioUrl")
    private String audioUrl;
    @JsonProperty("createdAt")
    private String createdAt;

    public Track() {}

    public String getId() { return id; }
    public void setId(String id) { this.id = id; }

    public String getTitle() { return title; }
    public void setTitle(String title) { this.title = title; }

    public String getArtist() { return artist; }
    public void setArtist(String artist) { this.artist = artist; }

    public String getGenre() { return genre; }
    public void setGenre(String genre) { this.genre = genre; }

    public int getDurationSeconds() { return durationSeconds; }
    public void setDurationSeconds(int durationSeconds) { this.durationSeconds = durationSeconds; }

    public String getCoverUrl() { return coverUrl; }
    public void setCoverUrl(String coverUrl) { this.coverUrl = coverUrl; }

    public String getSpotifyId() { return spotifyId; }
    public void setSpotifyId(String spotifyId) { this.spotifyId = spotifyId; }

    public String getAudioUrl() { return audioUrl; }
    public void setAudioUrl(String audioUrl) { this.audioUrl = audioUrl; }

    public String getCreatedAt() { return createdAt; }
    public void setCreatedAt(String createdAt) { this.createdAt = createdAt; }
}

File: backend/src/models/Track.js
const mongoose = require('mongoose');

const trackSchema = new mongoose.Schema({
  title: {
    type: String,
    required: true,
    trim: true,
  },
  artist: {
    type: String,
    required: true,
    trim: true,
  },
  genre: {
    type: String,
    required: true,
  },
  durationSeconds: {
    type: Number,
    default: 30,
  },
  coverUrl: {
    type: String,
    default: null,
  },
  spotifyId: {
    type: String,
    default: null,
  },
  audioUrl: {
    type: String,
    default: null,
  },
  createdAt: {
    type: Date,
    default: Date.now,
  },
}, {
  collection: 'tracks',
  timestamps: false,
  versionKey: false,
});

module.exports = mongoose.models.Track || mongoose.model('Track', trackSchema);
